### PR TITLE
[HIPIFY][doc] Cherry pick table revamp for docs/6.1.0

### PR DESCRIPTION
**Pull request by samjwu**
Cherry pick this PR https://github.com/ROCm/HIPIFY/pull/1410 into the 6.1.0 docs

### CODE REVIEW

**Review by LisaDelaney — APPROVED** [review]

**Review by emankov — CHANGES_REQUESTED** [review]
Fix https://github.com/ROCm/HIPIFY/pull/1410 is based on staging documentation (post 6.1.0). Here, you need to apply that fix on top of 6.1.0 w/o introducing documentation changes newer than 6.1.0.

**Comment by peterjunpark** [comment]
Hi @emankov, this PR should be in docs/6.1.0 and docs/6.1.1.

I checked the diffs -- the updates in #1410 are entirely cosmetic. It just converts the `md` file to `rst` and updates the navigation on ROCm docs to point to the correct file.

I'm reopening this as it also fixes broken download links in your docs. I guess these are better handled in `rst`.

Can you review please?

File: docs/hipify-clang.rst [hunk 1]
```diff
@@ -0,0 +1,809 @@
...
+    - \+
+    - \+
+  * - `17.0.1 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-17.0.1>`_,
+      `17.0.2 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-17.0.2>`_,
+      `17.0.3 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-17.0.3>`_,
+      `17.0.4 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-17.0.4>`_,
+      `17.0.5 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-17.0.5>`_,
+      `17.0.6 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-17.0.6>`_\*\*,
+      `18.1.0 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-18.1.0>`_,
+      `18.1.1 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-18.1.>`_
```
Comment (emankov) [inline docs/hipify-clang.rst:172]: 18.1. ?

Comment (emankov) [inline docs/hipify-clang.rst:172]: There were no any 18.x LLVM release at the time of shipping 6.1.0.

File: docs/hipify-clang.rst [hunk 2]
```diff
@@ -0,0 +1,809 @@
...
+``--clang-resource-directory=`` must be specified to point to the Clang resource directory, which is the
+parent directory for the ``include`` folder that contains ``__clang_cuda_runtime_wrapper.h`` and other
+header files used during the hipification process. For example:
+
+.. code:: shell
+
+  ./hipify-clang square.cu --cuda-path=/usr/local/cuda-12.3 --clang-resource-directory=/usr/llvm/18.1.1/dist/lib/clang/18
+
+The `Clang manual for compiling CUDA <https://llvm.org/docs/CompileCudaWithLLVM.html#compiling-cuda-code>`_
+may be useful.
```
Comment (emankov) [inline docs/hipify-clang.rst:232]: It is taken from 6.1.1: https://github.com/ROCm/HIPIFY/blob/amd-staging/CHANGELOG.md#hipify-for-rocm-611 and is not presented in 6.1.0. 

It looks like this change is not a cherry-pick.

**Review by emankov — CHANGES_REQUESTED** [review]
It looks like this change is not a cherry-pick.

File: docs/hipify-clang.rst (hunk 2 above)
Comment (peterjunpark) [reply docs/hipify-clang.rst:232]: Thanks @emankov. These content changes are already live on 6.1.0/latest [ROCm docs](https://rocm.docs.amd.com/projects/HIPIFY/en/latest/hipify-clang.html).

The additions you're talking about were made in #1404 and #1391 and are in latest. #1410 keeps the same content and just converts the Markdown file into rST.

I agree there is a discrepancy with the changelog. If this content is not appropriate for 6.1.0, let's fix it here?

Comment (emankov) [reply docs/hipify-clang.rst:232]: The contest is newer than 6.1.0, it is targeting 6.1.x. The issue is not only in the changelog, but in the 3rd parties SW support and features, introduce later than 6.1.0.

File: docs/hipify-clang.rst [hunk 3]
```diff
@@ -0,0 +1,809 @@
...
+    - \+
+    - \+
+  * - `17.0.1 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-17.0.1>`_,
+      `17.0.2 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-17.0.2>`_,
+      `17.0.3 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-17.0.3>`_,
+      `17.0.4 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-17.0.4>`_,
+      `17.0.5 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-17.0.5>`_,
+      `17.0.6 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-17.0.6>`_\*\*,
+      `18.1.0 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-18.1.0>`_,
+      `18.1.1 <https://github.com/llvm/llvm-project/releases/tag/llvmorg-18.1.>`_
```
Comment (emankov) [inline docs/hipify-clang.rst:172]: There were no 18.1.x for 6.1.0 yet...

Btw, the change doesn't look to me like a cherry-pick due to typos like "**18.1.**"

File: docs/hipify-clang.rst (hunk 2 above)
Comment (emankov) [reply docs/hipify-clang.rst:232]: `--clang-resource-directory` is not presented in 6.1.0, it is onlyt going to be presented in 6.1.x.

**Review by emankov — CHANGES_REQUESTED** [review]
Once again, it is not a cherry-pick for 6.1.0. The documentation mostly refers to the upcoming 6.1.x.

**Pull request closed without merging**